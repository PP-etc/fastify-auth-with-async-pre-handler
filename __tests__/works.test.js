const httpStatusCodes = require("http-status-codes");

describe("fastify-auth", () => {
  it("works", async done => {
    const fastify = require("fastify")();

    fastify
      .decorate("verifyJWTandLevel", function(request, reply, done) {
        // your validation logic
        done(); // pass an error if the authentication fails
      })
      .register(require("fastify-auth"))
      .after(() => {
        fastify.route({
          method: "POST",
          url: "/auth-multiple",
          preHandler: fastify.auth([
            fastify.verifyJWTandLevel
          ]),
          handler: (req, reply) => {
            req.log.info("Auth route");
            reply.send({ hello: "world" });
          }
        });
      });
    const response = await fastify.inject({
      method: "POST",
      url: "/auth-multiple"
    });
    expect(response.statusCode).toBe(httpStatusCodes.OK);
    expect(response.payload).toMatchSnapshot();
    done();
  });
});
