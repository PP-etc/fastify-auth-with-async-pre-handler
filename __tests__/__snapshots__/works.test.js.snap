// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fastify-auth works 1`] = `"{\\"hello\\":\\"world\\"}"`;
